### PR TITLE
[TTAHUB-3309] Only one connection per objective to ARO

### PR DESCRIPTION
**Pull request by GarrettEHill**
## Description of change

- Fix hole that created duplicate ARO entries when a file was uploaded
- migration to merge duplicate aro records for report from support case: ActivityReportId: 46116 

## How to test

1. On a report with and objective
2. Save
3. Add a file
4. Save
5. Add a file
6. Check the DB to confirm only a single ARO is on the report

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-0

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Review by hardwarehuman — APPROVED** [review]
Some comments in the migration, but otherwise good to go.

File: src/migrations/202408015000000-remove-dup-aros-via-uploadHandler.js [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+            FROM temp_reduced_aro s
+            WHERE o.id = s.id
+            AND o."objectiveId" = s."objectiveId"
+            AND o."activityReportId" = s."activityReportId"
+            RETURNING o.*
+        )
+        SELECT * FROM updated_rows;
+
+        -- Insert non-matching records into ActivityReportObjectives
+        CREATE TEMP TABLE temp_inserted_objectives AS
```
[the remark this answers is not in the document]
Comment (GarrettEHill) [reply src/migrations/202408015000000-remove-dup-aros-via-uploadHandler.js:399]:
```suggestion
        -- There should never be an INSERT performed here, but the code is present to complete the structure UID structure 
        CREATE TEMP TABLE temp_inserted_objectives AS
```

File: src/migrations/20240815000000-remove-dup-aros-via-uploadHandler.js [hunk 2]
```diff
@@ -0,0 +1,543 @@
+const { prepMigration, removeTables } = require('../lib/migration');
+
+module.exports = {
+  up: async (queryInterface) => queryInterface.sequelize.transaction(
+    async (transaction) => {
+      await prepMigration(queryInterface, transaction, __filename);
+
+      await queryInterface.sequelize.query(/* sql */`
+        -- This dedupes the AROs for one AR (46116) in an active support case
```
Comment (hardwarehuman) [inline src/migrations/20240815000000-remove-dup-aros-via-uploadHandler.js:9]:
```suggestion
        -- This dedupes the AROs for three ARs (46116, 46266, 45907) in active support cases
```

File: src/migrations/20240815000000-remove-dup-aros-via-uploadHandler.js [hunk 3]
```diff
@@ -0,0 +1,543 @@
...
+            WHERE o.id = s.id
+            AND o."objectiveId" = s."objectiveId"
+            AND o."activityReportId" = s."activityReportId"
+            RETURNING o.*
+        )
+        SELECT * FROM updated_rows;
+
+        -- Insert non-matching records into ActivityReportObjectives
+        -- There should never be an INSERT performed here, but the code is present to complete the structure UID structure
+        -- There should never be an INSERT performed here, but the code is present to complete the structure UID structure
```
Comment (hardwarehuman) [inline src/migrations/20240815000000-remove-dup-aros-via-uploadHandler.js:400]: Looks like a duplicate line?

Comment (GarrettEHill) [reply src/migrations/20240815000000-remove-dup-aros-via-uploadHandler.js:400]:
```suggestion
```

**Pull request merged**